riotdocker-base/build.sh

```diff
@@ -0,0 +1,77 @@
+#!/bin/bash
+
+# Automatically exit on error
+set -e
+
+COUNTER_STEP=0
+COUNTER_SUBSTEP=0
+BLUE="\e[34m"
+BOLD="\e[1m"
+NORMAL="\e[0m"
+
+step() {
+    COUNTER_SUBSTEP=0
+    COUNTER_STEP=$(("$COUNTER_STEP" + 1))
+    printf "${BLUE}${BOLD}==>${NORMAL}${BOLD} Step %d:${NORMAL} %s\n" "$COUNTER_STEP" "$1"
+}
+
+substep() {
+    COUNTER_SUBSTEP=$(("$COUNTER_SUBSTEP" + 1))
+    printf "${BLUE}${BOLD}    -->${NORMAL}${BOLD} Step %d.%d:${NORMAL} %s\n" \
+        "$COUNTER_STEP" "$COUNTER_SUBSTEP" "$1"
+}
+
+step_install_dev_tools() {
+    step "Installing development tools"
+    substep "Updating package index"
+    apt-get update
+
+    substep "Installing GCC"
+    apt-get -y --no-install-recommends install gcc
+
+    substep "Installing git"
+    apt-get -y --no-install-recommends install git
+
+    substep "Installing Python"
+    apt-get -y --no-install-recommends install \
```

Comment:
Doing multiple apt-get install in separate steps incurs some run-time cost from hooks being run after each apt-get.

But fine with me -- it's little time spent in CI, and much better readability gained.